.github/workflows/ci.yaml

```diff
@@ -0,0 +1,165 @@
+name: Build and Deploy the API
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+jobs:
+  test:
+    name: Run tests
+    runs-on: ubuntu-latest
+    env:
+      CI: "true"
+    steps:
+    - id: checkout
+      name: Check out source code
+      uses: actions/checkout@v2
+
+    - id: cache-seek
+      name: Check for npm cache hit
+      uses: c-hive/gha-npm-cache@v1
+
+    - id: install
+      name: Install dependencies
+      run: |-
+        npm ci
+
+    - id: test
+      name: Run unit tests
+      run: |-
+        npm test
+      env:
+        AWS_DEFAULT_REGION: 'eu-west-1'
+
+  build-docker:
+    name: Build Docker container
+    runs-on: ubuntu-latest
+    needs: test
+    steps:
+      - id: checkout
+        name: Check out source code
+        uses: actions/checkout@v2
+
+      - id: ref
+        name: Format docker tag and repository name.
+        run: |-
+          # This will take a ref like `refs/heads/master`
+          # and turn it into `refs-heads-master`
+          REF_ID=$(echo $GITHUB_REF | sed 's/\//-/g')
+
+          # the final tag is something like:
+          # refs-heads-master-a4f8bc313dae
+          # this is what we push to ECR
+          echo "::set-output name=tag::$REF_ID-$GITHUB_SHA"
+
+          # This will take a GitHub repo name like `biomage-ltd/iac`
+          # and turns it into `iac`. This will be the name of the
+          # ECR repository.
+          IMAGE_REPO_NAME=$(echo $GITHUB_REPOSITORY | awk -F '/' '{print $2}')
+          echo "::set-output name=repo-name::$IMAGE_REPO_NAME"
+
+      - id: set-up-creds
+        name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-1
+
+      - id: login-ecr
+        name: Login to Amazon ECR
+        uses: aws-actions/amazon-ecr-login@v1
+      
+      - id: create-ecr-registry
+        name: Create an ECR repository (if needed)
+        # This will fail if the registry already exists, which is fine. If there is some other
+        # error, the `push` step will fail instead.
+        continue-on-error: true
+        run: |-
+          aws ecr create-repository --repository-name $REPO_NAME --image-tag-mutability IMMUTABLE
+        env:
+          REPO_NAME: ${{ steps.ref.outputs.repo-name }}
+
+      - id: build
+        name: Build Docker image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./Dockerfile
+          platforms: linux/amd64
+          tags: ${{ format('{0}/{1}:{2}', steps.login-ecr.outputs.registry, steps.ref.outputs.repo-name, steps.ref.outputs.tag) }}
+          push: false
+
+      - id: push
+        name: Push docker image to ECR
+        run: |-
```

Comment:
make it push the image only if there is a push to master branch

Reply:
Fixed.